### DISCUSSION TITLE
Anti-phishing Notification Feature - Please make opt-in only

### DISCUSSION THREAD

**Discussion by mwmason** [simple-login/app#877] [Feature Requests]
Recently,  I received a forwarded email via one my SL aliases that had been altered to be prefixed by a red text banner that read: "This email failed anti-phishing checks when it’s received by SimpleLogin, be careful with its content."

Honestly, it was a little disconcerting because I didn't know for sure that it had be altered by SimpleLogin.  I contacted customer support and was told that this is indeed a new feature.

While this may be of use and interest to some, I believe this is just stepping too far in terms of the intent of the forwarding service.    

Please make this an opt-in feature rather than the default behavior.

**Comment by thibaultmol** [comment]
I like the feature, but was also surprised by it.
I think from a security standpoint it makes sense to have this enabled by default.
But it would need more context as to WHY the mail was marked as phising.
Also would have been best if they atleast had communicated before hand "Hey, we'll be enabling a new feature starting next month where we'll say if an email might be suspect of phising and this is how you'll see that in action"

**Reply by nguyenkims** [reply]
Due to a security audit we've gone through recently, the phishing risk is quite important and we want to make sure this is handled properly before this info becomes public. That's why we decided to roll out this feature as soon as possible. In the next iteration, we'll make sure to make it optional (maybe on the alias level?) so you can turn it off.

**Reply by mwmason** [reply]
Were the recommendations of the audit that you should modify email content?  Is there no other header info where risk of phishing info could be flagged?  I think my reaction to this is based on the fact that there was no announcement, if feels intrusive, the email that was modified was actually a false positive - everything seems rushed.   I wasn't sure it even came from SL - not something I would think you'd want with your service.

What mechanism is being used to determine if it's a phishing attempt?  

I would like to opt out altogether - I can't see any benefit at the alias level.

**Reply by nguyenkims** [reply]
The detection isn’t based on the email content at all, it’s based on the IP address and the domain the email comes from so at no moment in the process, SimpleLogin analyses the email content.

We'll make sure to create a way to opt out of this feature in our next version.

**Comment by thibaultmol** [comment]
From what I understand from support. It's probably purely dmarc based.
The anti phishing system is triggered when the dmarc records aren't correct
(basically a way to verify that an email is indeed coming from a correct
source. Using dns records
https://youtu.be/qP9ODdimHvM

**Comment by thibaultmol** [comment]
I think this is very useful as a safety feature and personally would like
to keep it enabled.
It's just best if we got more transparency as to WHY something got flagged,
so that it gives us, the user, info on what we should check to verify if
the email is indeed legit

PS, congrats on the Proton(mail) acquisition. I actually messaged proton
last year saying that it's a shame they don't have an alias  system as
advanced and well done as SL.
Well that's fixed now I guess XD

Cheers,
Thibault

**Reply by nguyenkims** [reply]
Yes we'll make sure to add more info about this feature. Currently it's quite simple and based on dmarc check.

**Comment by marvin8** [comment]
As others have said... this is a good feature in principle and I'll likely continue using it when / if it becomes optional.

However I feel the introduction of the anti-phishing checks was rushed and not  well or at all communicated. The lack of communication before introduction of such a new and what feels invasive feature feels weird for an organization that claims to follow a 'user first' philosophy.

Anyway, congratulations on implementing something for phishing protection.

My wish list in regards to anti-phishing:

- Provide some kind of explanation why a particular email has failed the implemented anti-phishing checks.
- Make adding of the message in actual content of the email optional.
- Maybe include a specific email header instead / as well... maybe something like `X-SimpleLogin-Anti-Phishing: Failed`
- In the future, please communicate changes to your users **before** or at least at the time of activating major new features.

**Reply by nguyenkims** [reply]
Thanks for the idea! We'll implement the suggestions in the next version. And we'll make sure to inform users before rolling out the next important feature :).

**Comment by rowatt** [comment]
I'd like to echo the comments re. making this feature optional and communicating better before launching significant user impacting features. I'd also like to echo the congratulations for the Protonmail acquisition - that sounds like a really positive step!

To add to the discussion, I can say that the feature is completely useless for me. I am getting the message on a number of emails, most of which I have been receiving mails from for years. So much so, that if the message were genuinely flagging a phishing attempt I am likely to ignore it unless something else in the mail gives me cause to be suspicious. For me, it would not work as a warning with a well crafted phishing email.

I've also had a significant increase in mails getting blocked altogether - again mails which I have been happily receiving from for a long time.

So net, net - for me - the new security features are net negative. All false positives so far, intrusive and inconvenient - and therefore much less likely to have the intended impact when there is a bad email.

I am not against security related features, just make them opt in, or if not opt in, make sure they are very well tested and work in practice before rolling out for everyone.

@nguyenkims Is there any way behind the scenes you can turn this off on my account, or could you give a timeframe for when this can be a settings based opt-in or opt-out?

**Comment by thibaultmol** [comment]
While I understand your points @rowatt I think this is a good opportunity to see what email senders actually have badly configured email servers/accounts. 
The couple emails that have been either blocked or had the suffix message added at the top ,I have contacted and informed them about their DMARC settings not being correct. They're happy with the feedback, as other email systems also look at dmarc to verify if an email is real or potentially spam (not immediately block it, but def use it as one of many metrics to judge an email on spam or not)

**Reply by rowatt** [reply]
I agree that helping senders understand that their systems are misconfigured is a good thing. However, I don't think it's the responsibility of a SimpleLogin user to do that. Personally I don't have time to write back to all the lists where I have had this message, and in some cases it's not even clear who I would write to.

Is there a way where the SimpleLogin system could automatically send a message back to people with misconfigured DMARC (or anything else)? That would be a great service, and could help spread awareness of SimpleLogin (and ProtonMail).

**Comment by mwmason** [comment]
Looks like Protonmail has been providing this check for a while - https://protonmail.com/blog/prevent-phishing-attacks/ .  And as I suspected, there's a non-obtrusive UI element in the email summary of PM that alerts you to the failed DMARC check.

While I realize not all your users are also PM users, it further supports the ask to make this an optional feature - since it would be redundant for PM users.

**Reply by nguyenkims** [reply]
As the email is actually sent from SimpleLogin server, the current anti phishing measure provided by an email provider doesn't work. SimpleLogin wants to provide the same protection for every email provider. That being said, we might add in the future dedicated integration in ProtonMail that better mitigate the phishing risk.

**Reply by jmreekes** [reply]
I have dmarc enabled on my domains per the instructions. I was wondering if anything would break if I were to add additional items like fo=1, rua address, ruf address to my DMARC Record on my SimpleLogin domains

**Reply by nguyenkims** [reply]
No everything will work fine if you have SPF and DKIM correctly set up.

**Comment by trikotret** [comment]
Its almost going to 4 months now. Any idea when you would have the option to opt out for an alias? I used SL for work related instead of Google workspace. I don't want to reply to someone and accidently have SL verbiage about anti phishing.

**Reply by nguyenkims** [reply]
@trikotret with this option disabled, there's no way to inform users about the phishing risk. Usually this can be fixed easily on the sender side (they can fix their DMARC or just remove DMARC) and I suggest to ask the sender to fix it. Their emails can also be put into Spams anywhere else.

**Reply by trikotret** [reply]
Its from T-Mobile Corp. No way I can get them to make changes on their end.

**Reply by rowatt** [reply]
> there's no way to inform users about the phishing risk.

What about users who don't want to be informed? Since you enabled this feature I have received 100's of mails warning me of phishing risk, but not 1 was phishing or spam - with so many false positives, the chance of me taking notice of the warning in a real case are much diminished. All my mails are routed to Google Workspace, which already does an excellent job of finding these mails and sending them to spam.

> Usually this can be fixed easily on the sender side (they can fix their DMARC or just remove DMARC) and I suggest to ask the sender to fix it.

As trikotret points out, this just isn't practical in many cases.

I love the Simple Login service in every respect except this one, but this issue is a big deal for me - I am actively looking for alternatives and will cancel my subscription if I find a viable alternative. Turning it off for specific aliases would be excellent (and completely solve the too many false positives issue for me), but just being able to turn it off globally would be a good start.

**Reply by trikotret** [reply]
> @trikotret with this option disabled, there's no way to inform users about the phishing risk. Usually this can be fixed easily on the sender side (they can fix their DMARC or just remove DMARC) and I suggest to ask the sender to fix it. Their emails can also be put into Spams anywhere else.

Is there a way to not include this warning in the reply email? I can delete before sending but sometimes Ill forget. It'll be better if it doesnt get included with replies.

**Reply by JtwoA** [reply]
> I am actively looking for alternatives and will cancel my subscription if I find a viable alternative.

That seems a bit drastic.  Feels like you were looking for a way out already and this is just giving you the excuse.

**Comment by weiw11** [comment]
@nguyenkims any update on this? There shouldn't be a reason for SL to add notification messages on the email body for reasons outside user's control.

**Reply by trikotret** [reply]
ya this is really annoying. Unfortunately, I can't tell T-Mobile, Microsoft, Starbucks, etc. etc. to fix their email. I do have lots of emails going back and forth with T-Mobile and I accidentally slipped a few red warnings.

**Comment by trikotret** [comment]
Anyway, to add this Anti-phishing red notification under our email address and not in email body? Thats how proton adds its header "This email has failed its domain's authentication requirements. It may be spoofed or improperly forwarded." This way we can reply to email and not have to delete that red warning.

**Comment by nolith** [comment]
Any chance we could at least disable it for mailing lists?

Every message I receive from open source mailing list is flagged as phishing. The mail server adds a footer to each message invalidating the signature of the message.

I've asked about implementing some of the mitigations described in https://docs.mailman3.org/projects/mailman/en/latest/src/mailman/handlers/docs/dmarc-mitigations.html but I was told the ML runs on mailman v2 that does not support those features.

**Comment by ArcticWarmth** [comment]
If we could turn this off on a per-alias basis (or at all) that would be great. I forward emails from a google account that I do not manage to an alias every email I get is marked as phishing. Having no way to turn this off or at least remove the subject line/move warning to bottom of email is problematic.

**Comment by hiccups1980** [comment]
The problem is that the anti-phishing function incorrectly recognizes many emails (at least for me) as phishing. It would be better to have a function where I can say in simplelogin:
"yes, I know the sender. mark him as trustworthy."

**Comment by TobiasAT** [comment]
It’s really frustrating that nothing has improved in more than three years. I agree with the others, there should be an option to mark a sender as safe so it’s not flagged as a “possible phishing attempt.” 
I also don’t see it as my responsibility to tell senders to fix their DMARC records for SimpleLogin. I suspect they’ll end up blocking SL addresses before they fix those records.

**Comment by marvin8** [comment]
Yep, it's been one these minor frustrations, among some others that have made me look for this type of service somewhere else. You can now disregard all my previous comments (if you wish) as I am no longer a user.

**Comment by benjuade** [comment]
Check also this idea: https://github.com/simple-login/app/discussions/2182

**Comment by selflesssloth** [comment]
I just want to give this topic another "Thumbs up"
I have an alias for a classical mailing list so every email there is "classically forwarded", what results in every email from that mailing list to get quarantined.
Please make this feature Opt-Out or Opt-In on an Alias level!

**Comment by brlumen** [comment]
Imagine how many crooked emails there are in the world! And it depends only on the specialists who are responsible for their mail service! Shame!
And to the point: How do you detect phishing if an email has phishing content (context) but is sent from a properly configured domain, Let's say you just picked up a password and sent a phishing email from a compromised account? In fact, phishing is more about the intent (the content of the email) rather than the authorization of the sender (which is what DMARC is responsible for.) For a failed DMARC check, it is more reasonable to warn that the sender is possibly spoofed rather than that it is phishing.

**Comment by Dubz** [comment]
How is it that I create a new email alias, the website I was on when I created it sent me an email, and it was immediately flagged for "Possible phishing attempt"? The entire purpose of this is "burner" or site specific emails. Now if I got an email from Mohammad with an offer for their family inheritance I'd understand, but the same site? Really?

On top of this, there's no "report false positive" option anywhere. So we're just supposed to take this as-is with no control?

**Comment by huntfx** [comment]
I keep checking back to this as it's the _one_ feature of SL that I really dislike. 

Half of the emails forwarded from work get stamped with `[Possible phishing attempt]` which is useless when I simply want to back up my work emails to Proton. Every now and then I'll get "An email has been quarantined". Fortunately those have not yet been anything important so it's been more of an annoyance than anything else.

I literally just want a copy of the email in my inbox, that's it. My Gmail address at work has spam detection, let me figure out if anything slips through, I'd prefer 1 false negative than the hundreds of false positives.

Surely this is a low hanging fruit that would take just a couple of hours to implement? A toggle for each alias, linked to a database column that indicates if its enabled or not, which is checked before the phishing code is run.

I know it'll be slightly more complex than this, but it took me 10 mins to find where to edit the code.
```python
# app/app/models.py
class Alias(Base, ModelMixin):
    ...
    phishing_check_enabled = sa.Column(sa.Boolean(), default=True, nullable=False)

# app/handler/dmarc.py
def apply_dmarc_policy_for_forward_phase(
    ...
    if not alias.phishing_check_enabled:
        return msg, None

# templates/dashboard/index.html
# add a "custom-switch cursor" somewhere as the physical toggle

# static/js/index.js
async function disableAliasPhishingCheck(aliasId, alias) {
   ... 
   
# app/api/views/alias.py
@api_bp.route("/aliases/<int:alias_id>/toggle", methods=["POST"])
@require_api_auth
def toggle_alias_phishing_check(alias_id):
    alias_utils.change_alias_status(
        alias,
        phishing_check_enabled=not alias.phishing_check_enabled,
        message=f"Set phishing_check_enabled={not alias.phishing_check_enabled} via API",
    )
```
